apps/web/src/app/(pages)/(dashboard)/roles/page.tsx

```diff
@@ -63,16 +66,23 @@ export default function Roles({
   }, [toast, mounted]);
 
   const reviews = api.review.list.useQuery({
+    search: searchParams?.search,
     options: validationResult.success ? validationResult.data : {},
   });
 
   const [selectedReview, setSelectedReview] = useState<ReviewType | undefined>(
     reviews.data ? reviews.data[0] : undefined,
   );
 
+  useEffect(() => {
```

Comment:
in general we want to avoid this kind of `useEffect`, but I'm assuming there was some kind of bug due to the fact that reviews are updated based on the child `SearchFilter` component?